plugins/dns_service/app/controllers/dns_service/request_zone_wizard_controller.rb

```diff
@@ -8,10 +8,20 @@ class RequestZoneWizardController < ::DashboardController
     def new
       @zone_request = ::DnsService::ZoneRequest.new(nil)
       @pools = cloud_admin.dns_service.pools[:items]
+      domain_config = DomainConfig.new(@scoped_domain_name)
+      disabled_providers = domain_config.disabled_dns_providers?
 
-      # this needs to be removed when migration is done
+      # filter disabled DNS providers
       @pools.reject! do |pool|
-        pool.attributes["attributes"]["label"] == "New External SAP Hosted Zone"
+        pool_label = pool.attributes["attributes"]["label"]
+        is_disabled = false
+        is_disabled = disabled_providers.include?(pool_label) if disabled_providers
```

Comment:
This line would raise an error if:

```ruby
disabled_dns_providers: 123 → 123.include? → raises NoMethodError
disabled_dns_providers: true → true.include? → raises NoMethodError
```

I would catch the error and return false

```ruby
is_disabled = begin
  disabled_providers.include?(pool_label)
rescue NoMethodError
  false
end
```